libgeopmd/src/RuntimeStats.cpp

```diff
@@ -0,0 +1,181 @@
+        double result = NAN;
+        if (m_moments[metric_idx].count != 0) {
+            result = m_moments[metric_idx].m_1 /
+                     m_moments[metric_idx].count;
+        }
+        return result;
+    }
+
+    double RuntimeStats::std(int metric_idx) const
+    {
+        check_index(metric_idx, __func__, __LINE__);
+        double result = NAN;
+        if (m_moments[metric_idx].count > 1) {
+            result = std::sqrt(
+                         (m_moments[metric_idx].m_2 -
+                          m_moments[metric_idx].m_1 *
+                          m_moments[metric_idx].m_1 / m_moments[metric_idx].count) /
+                         (m_moments[metric_idx].count - 1));
+        }
+        return result;
+    }
+
+    double RuntimeStats::skew(int metric_idx) const
+    {
+        throw Exception("RuntimeStats::" + std::string(__func__) + " not yet implemented",
+                        GEOPM_ERROR_NOT_IMPLEMENTED, __FILE__, __LINE__);
+    }
+
+    double RuntimeStats::kurt(int metric_idx) const
+    {
+        throw Exception("RuntimeStats::" + std::string(__func__) + " not yet implemented",
+                        GEOPM_ERROR_NOT_IMPLEMENTED, __FILE__, __LINE__);
+    }
+
+    double RuntimeStats::lse_linear_0(int metric_idx) const
+    {
+        throw Exception("RuntimeStats::" + std::string(__func__) + " not yet implemented",
+                        GEOPM_ERROR_NOT_IMPLEMENTED, __FILE__, __LINE__);
+    }
+
+    double RuntimeStats::lse_linear_1(int metric_idx) const
+    {
+        throw Exception("RuntimeStats::" + std::string(__func__) + " not yet implemented",
+                        GEOPM_ERROR_NOT_IMPLEMENTED, __FILE__, __LINE__);
+    }
+
+    void RuntimeStats::reset(void)
+    {
+        for (auto &it : m_moments) {
+            it.count = 0;
+            it.m_1 = 0.0;
+            it.m_2 = 0.0;
+            it.m_3 = 0.0;
+            it.m_4 = 0.0;
+        }
+    }
+
+    void RuntimeStats::update(const std::vector<double> &sample)
+    {
+        if (sample.size() != m_moments.size()) {
+            throw Exception("RuntimeStats::update(): invalid input vector size: " + std::to_string(sample.size()),
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        auto moments_it = m_moments.begin();
+        for (const auto &ss : sample) {
+            if (!std::isnan(ss)) {
+                moments_it->count += 1;
+                if (moments_it->count == 1) {
+                    moments_it->first = ss;
+                    moments_it->min = ss;
+                    moments_it->max = ss;
+                }
+                moments_it->last = ss;
+                if (moments_it->min > ss) {
+                    moments_it->min = ss;
+                }
+                if (moments_it->max < ss) {
+                    moments_it->max = ss;
+                }
+                double mm = ss;
+                moments_it->m_1 += mm;
```

Comment:
I guess `m_#` is (catching up on wikipedia just now) is a running accumulation of `count * #th raw sample moment`